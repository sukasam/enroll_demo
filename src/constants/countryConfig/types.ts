import { Alpha2, Alpha3, Currency, optInNotificationTypes } from "./enums";

export type Language = {
    code: string;
    name: string;
};

export interface PackPerk {
    icon: string;
    perk: string;
}

export interface PackExtra {
    icon: string;
    title: string;
    description: string;
}

export interface PackProduct {
    image: string;
    title: string;
    description: string;
}

export interface VariantOption {
    // Id within option to distinguish between values. Each option has its own set of ids.
    interOptionId: number;
    title: string;
    description?: string;
    disclaimer?: string;
    limitedColor?: string;
    image?: string;
}

export interface Variant {
    sku: string;
    name: string;
    description: string;
    productImage: any;
    partsCount: string;
    tabImage: any;
    options: {
        [key: string]: VariantOption;
    };
    price: number;
    rrp: number;
}

export interface DigitalPack {
    name: string;
    isDigital: true;
    buttonName: string;
    description: string;
    showPaymentType: boolean;
}

export interface CompleteDigitalPack extends DigitalPack {
    variants: Variant[];
}

export interface Pack {
    mainFeatures: string[];
    accentColor: string;
    buttonName: string;
    kitDescription: string;
    backgroundColor: string;
    perks: PackPerk[];
    extras: PackExtra[];
    products?: PackProduct[];
    annotations: string;
    showPaymentType: boolean;
    isDigital: false;
}

export interface CompletePack extends Pack {
    variants: Variant[];
}

export type ProductConfig = {
    recommendedPack: number | null;
    packs: (Pack | DigitalPack)[];
};

export type CompleteProductConfig = {
    recommendedPack: number;
    packs: (CompletePack | CompleteDigitalPack)[];
};

export interface WillCallAddress {
    address1?: string;
    address2?: string;
    city?: string;
    state?: string;
    zip?: string;
    country?: string;
    mapLink?: string;
    mapLinkKey?: string;
    rosettaKey?: string;
}

export type PaymentOptions = {
    hasPayPal?: boolean;
    hasCash?: boolean;
    hasBankWire?: boolean;
    hasCreditCard?: boolean;
    hasSavedCreditCard?: boolean;
    hasKlarna?: boolean;
};

export type MarketExceptions = {
    zipLength?: number | number[] | null;
    zipPattern?: RegExp | null;
    taxIdPattern?: RegExp;
    secondaryTaxIdPattern?: RegExp;
    taxInclusiveFrontEnd?: boolean;
    taxInclusiveBackEnd?: boolean;
    hasSecondaryTaxId?: boolean;
    secondaryTaxIdData?: {
        label: string;
    }[];
    disableTaxId?: boolean;
    hiddenShippingProvince?: boolean;
    useQuotesToValidateAddress?: boolean;
    hasCommissionPaymentOptions?: boolean;
    showOnboardingGuide?: boolean;
    showVisaAgreement?: boolean;
    hiddenShippingQuotesCity?: boolean;
    hiddenShippingQuotesZip?: boolean;
    willCallAddresses?: WillCallAddress[];
    thankYouImage?: string;
    hideUnicityPhone?: boolean;
    paymentOptions?: PaymentOptions;
    showStatesField?: boolean;
    useWorldPay?: boolean;
    directWorldPay?: boolean;
    usePayPal3ds?: boolean;
    notificationType?: optInNotificationTypes | undefined;
    onlyHasWillCall?: boolean;
    showOnlineRegistration?: boolean;
    hiddenShippingCity?: boolean;
    hiddenTooltip?: boolean;
    hiddenDecimal?: boolean;
    updateMainAddress?: boolean;
    showMemberApplicationGuide?: boolean;
    useGlobalFormatCurrency?: boolean;
    hiddenEnrollerId?: boolean;
    hiddenPersonalData?: boolean;
    useCookie?: boolean;
    formCreateAccount?: string[];
    formUpdateAccount?: string[];
};

type BaseCountryConfig = {
    alpha3: Alpha3;
    alpha2: Alpha2;
    name: string;
    languages: { code: string; name: string; default: boolean }[];
    marketExceptions: MarketExceptions;
};

export enum PaymentLogo {
    Visa = "visa",
    Mastercard = "mastercard",
    Amex = "amex",
    Discover = "discover",
    PayPal = "paypal",
    JCB = "jcb"
}

export type SupportedCountryConfig = {
    isSupported: true;
    productConfig: ProductConfig;
    currency: Currency | null;
    paymentLogos: PaymentLogo[];
    dateFormat: "YMD" | "DMY" | "MDY";
    phonePrefix?: string;
    alwaysShowGuide?: boolean;
    isNoPurchaseMarket?: boolean;
    addressFormat?: string;
} & BaseCountryConfig;

export type UnsupportedCountryConfig = Omit<
    Partial<SupportedCountryConfig>,
    "isSupported"
> & {
    isSupported: false;
    redirect?: string;
} & BaseCountryConfig;

export type CountryConfig = SupportedCountryConfig | UnsupportedCountryConfig;

type OmittedProps =
    | "alpha2"
    | "alpha3"
    | "name"
    | "phonePrefix"
    | "isSupported"; // List the properties to omit
export type TemplateSupportedCountryConfig = {
    [K in keyof SupportedCountryConfig as K extends OmittedProps
        ? never
        : K]: SupportedCountryConfig[K];
};

export type Locale = {
    countryCode: Alpha2 | null;
    language: string | null;
};
