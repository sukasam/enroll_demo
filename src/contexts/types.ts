import { Alpha2 } from "Constants/countryConfig/enums";
import { Product } from "types/Product";
import { OrderResult } from "./types/OrderContextTypes";

export type Settings = {
    locale: string | null;
    version: string;
    cachedCatalogLocale: string | null;
};

type BaseUser = {
    firstName?: string;
    lastName?: string;
    email: string;
    birthDate: string;
    fullName: string;
    phoneNumber: string;
};

export type EnrollmentUser = BaseUser & {
    taxId?: string;
    secondaryTaxId?: string;
    address1: string;
    address2: string;
    city: string;
    state: string;
    zip: string;
    country: Alpha2;
    password: string;
    coapplicantFirstName: string;
    coapplicantLastName: string;
    newsletter: boolean;
    enroller: string | null;
    sponsor: string | null;
    commission: "paylution" | "direct_deposit";
    iban: string;
    bankName: string;
    accountHolder: string;
    bic: string;
};

export type UpgradeUser = BaseUser & {
    id: string;
    href: string;
    email: string;
    shippingAddress: object;
    type: string;
};

export type AppContextType = {
    loading: boolean;
    setLoading: (value: boolean) => void;
    locale: string | null;
    setLocale: (value: string) => void;
    version: string;
    selectedSku: string | null;
    setSelectedSku: (value: string) => void;
    enrollerId: string | null;
    setEnrollerId: (value: string | null) => void;
    sponsorId: string | null;
    setSponsorId: (value: string | null) => void;
    products: Product[];
    setProducts: (value: Product[]) => void;
    cachedCatalogLocale: string | null;
    setCachedCatalogLocale: (value: string) => void;
    userData: EnrollmentUser | UpgradeUser | null;
    setUserData: (value: EnrollmentUser | UpgradeUser) => void;
    userToken: string;
    setUserToken: (token: string) => void;
    orderResult: OrderResult | null;
    setOrderResult: (result: OrderResult) => void;
};
