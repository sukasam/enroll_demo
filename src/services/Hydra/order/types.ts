import { Alpha2 } from "Constants/countryConfig/enums";
import { Address } from "types/Address";
import { CustomerType } from "types/enums";

export enum TransactionType {
    AuthorizeAndCapture = "AuthorizeAndCapture",
    Capture = "Capture",
    IOU = "IOU"
}

export enum TransactionMethod {
    CreditCard = "CreditCard",
    CreditCardAlias = "CreditCardAlias",
    BankDraft = "BankDraft",
    BankWire = "BankWire",
    Cash = "Cash",
    PayPal = "PayPal",
    EWalletToken = "EWalletToken",
    SavedCreditCard = "SavedCreditCard",
    PayPal3ds = "Trust-PayPal",
    Klarna = "Klarna"
}

export type HumanName = {
    firstName?: string;
    lastName?: string;
    "fullName@en"?: string;
    fullName?: string;
};

export type BasePayment = HumanName & Partial<Address>;

export type BuildPaymentCC = BasePayment & {
    method: TransactionMethod.CreditCard;
    type: TransactionType.AuthorizeAndCapture;
    creditCardNumber: string;
    expiry: string;
    cvc: string;
    sessionId: string | never;
    bankName?: string;
};

export type BuildPaymentBankDraft = BasePayment & {
    method: TransactionMethod.BankDraft;
};

export type BuildPaymentBankWire = BasePayment & {
    method: TransactionMethod.BankWire;
};

export type BuildPaymentCash = BasePayment & {
    method: TransactionMethod.Cash;
    bankName?: string;
};

export type BuildPaymentPayPal = BasePayment & {
    method: TransactionMethod.PayPal;
};

export type BuildPaymentPayPal3ds = BasePayment & {
    method: TransactionMethod.PayPal3ds;
};

export type BuildPaymentKlarna = BasePayment & {
    method: TransactionMethod.Klarna;
};

export type BuildPayment =
    | BuildPaymentCC
    | BuildPaymentBankDraft
    | BuildPaymentBankWire
    | BuildPaymentCash
    | BuildPaymentPayPal
    | BuildPaymentPayPal3ds
    | BuildPaymentKlarna;

export type BuildShipping = BasePayment & {
    phone?: string;
    email: string;
};

export type CustomerObject = {
    mainAddress: Address;
    humanName: HumanName;
    type: CustomerType;
    taxTerms?: {
        taxId: string;
    };
    signature: {
        value: string;
    };
};

export type BuildOrderItem = {
    sku: string;
    qty: number;
};

export type BuildPayPal = BasePayment & {
    method: TransactionMethod.PayPal | TransactionMethod.PayPal3ds;
    ewalletToken: string;
    nonce: string;
    usePayPal3ds: boolean;
};

export type AddTransactionsProps = {
    payment: BuildPayment;
    paypal?: BuildPayPal;
    worldPaySessionId?: string;
};

export type BaseTransactionItem = {
    amount: string;
    type: TransactionType;
    method: TransactionMethod;
    billToEmail?: string;
    billToPhone?: string;
    billToAddress?: Address;
};

export type CreditCardTransactionItem = BaseTransactionItem & {
    method: TransactionMethod.CreditCard; // | TransactionMethod.CreditCardAlias;
    methodDetails: {
        payer: string;
        creditCardNumber?: string;
        alias?: string;
        creditCardExpires?: string;
        creditCardSecurityCode?: string;
        bankName?: string;
        branchName?: string;
    } & (
        | { sessionId: string; bankName: string; challengeWindowSize: string }
        | { sessionId?: never; challengeWindowSize?: never }
        | { bankName?: string }
    );
};

export type BankTransactionItem = BaseTransactionItem & {
    method: TransactionMethod.BankDraft | TransactionMethod.BankWire;
};

export type CashTransactionItem = BaseTransactionItem & {
    method: TransactionMethod.Cash;
    methodDetails: {
        bankName?: string;
    };
};

export type PayPalTransactionItem = BaseTransactionItem & {
    method: TransactionMethod.EWalletToken;
    methodDetails: {
        bankName: TransactionMethod.PayPal;
        ewalletToken: string;
        payer: string;
    };
};

export type PayPal3dsTransactionItem = BaseTransactionItem & {
    type: TransactionType.IOU;
    method: TransactionMethod.CreditCard;
    methodDetails: {
        bankName: TransactionMethod.PayPal3ds;
        payer: string;
    };
};

export type KlarnaTransactionItem = BaseTransactionItem & {
    method: TransactionMethod.Klarna;
    methodDetails: {
        payer: string;
        bankName: string;
        branchName: string;
        successURL?: string;
        pendingURL?: string;
        failureURL?: string;
        cancelURL?: string;
    };
};

export type TransactionItem =
    | CreditCardTransactionItem
    | BankTransactionItem
    | CashTransactionItem
    | PayPalTransactionItem
    | PayPal3dsTransactionItem
    | KlarnaTransactionItem;

export type OrderObject = {
    lines: {
        items: {
            item: {
                href: string;
            };
            quantity: number;
        }[];
    };
    shipTo: BuildShipping;
    customer: CustomerObject;
    shippingMethod: {
        href: string;
    };
    transactions: {
        items: TransactionItem[];
    };
    source: {
        agent: "Enroll";
        platform: string;
        medium: "Internet";
        version: string;
        market: Alpha2;
    };
};
