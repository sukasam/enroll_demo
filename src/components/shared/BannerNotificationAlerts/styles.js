import { css } from "@emotion/react";
import theme from "Styles/theme";

export default css`
    .container {
        max-width: calc(${theme.desktopWidth}px + 48px);
        display: auto;
        width: 100%;
        margin: 0 auto;
        padding: 0 24px;
    }

    .banner {
        max-width: ${theme.desktopWidth}px;
        display: flex;
        align-items: center;
        justify-content: center;
        flex-direction: column;

        background: #fef8e2; /* Yellow background */
        padding: 20px; /* Increased padding for better scaling */
        margin: 10px 0; /* Add some space around the banner */
        box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
        border-radius: 12px;
    }

    .message {
        font-family: "Inter";
        font-style: normal;
        font-weight: 400;
        font-size: 16px;
        line-height: 23px;
        color: #955100;
    }

    .children {
        margin-top: 10px;
    }
`;
