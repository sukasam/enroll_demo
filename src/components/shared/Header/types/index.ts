export type HeaderProps = {
    hideLocale: boolean;
    referralBar: boolean;
};
