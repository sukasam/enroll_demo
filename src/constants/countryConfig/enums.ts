export enum Alpha3 {
    USA = "USA",
    PRI = "PRI",
    CAN = "CAN",
    GBR = "GBR",
    ARE = "ARE",
    AIA = "AIA",
    AUT = "AUT",
    AUS = "AUS",
    BRB = "BRB",
    BEL = "BEL",
    BHR = "BHR",
    BES = "BES",
    BHS = "BHS",
    CHE = "CHE",
    COL = "COL",
    CUW = "CUW",
    CZE = "CZE",
    DEU = "DEU",
    DNK = "DNK",
    DMA = "DMA",
    DOM = "DOM",
    DZA = "DZA",
    EGY = "EGY",
    ESP = "ESP",
    FRA = "FRA",
    GRD = "GRD",
    HUN = "HUN",
    IRL = "IRL",
    IND = "IND",
    ITA = "ITA",
    JOR = "JOR",
    JPN = "JPN",
    KEN = "KEN",
    KOR = "KOR",
    KWT = "KWT",
    LBN = "LBN",
    LUX = "LUX",
    MEX = "MEX",
    NLD = "NLD",
    NOR = "NOR",
    NZL = "NZL",
    OMN = "OMN",
    PHL = "PHL",
    POL = "POL",
    SAU = "SAU",
    SWE = "SWE",
    SGP = "SGP",
    TCA = "TCA",
    THA = "THA",
    TUR = "TUR",
    TTO = "TTO",
    UKR = "UKR",
    VIR = "VIR",
    MAR = "MAR",
    LAO = "LAO",
    MYS = "MYS",
    MMR = "MMR",
    TWN = "TWN",
    VNM = "VNM",
    HKG = "HKG",
    KHM = "KHM",
    IDN = "IDN",
    JAM = "JAM",
    AFG = "AFG",
    ALA = "ALA",
    ALB = "ALB",
    ASM = "ASM",
    AND = "AND",
    AGO = "AGO",
    ATA = "ATA",
    ATG = "ATG",
    ARG = "ARG",
    ARM = "ARM",
    ABW = "ABW",
    AZE = "AZE",
    BGD = "BGD",
    BLR = "BLR",
    BLZ = "BLZ",
    BEN = "BEN",
    BMU = "BMU",
    BTN = "BTN",
    BOL = "BOL",
    BIH = "BIH",
    BWA = "BWA",
    BVT = "BVT",
    BRA = "BRA",
    IOT = "IOT",
    BRN = "BRN",
    BGR = "BGR",
    BFA = "BFA",
    BDI = "BDI",
    CMR = "CMR",
    CPV = "CPV",
    CYM = "CYM",
    CAF = "CAF",
    TCD = "TCD",
    CHL = "CHL",
    CHN = "CHN",
    CXR = "CXR",
    CCK = "CCK",
    COM = "COM",
    COK = "COK",
    CRI = "CRI",
    CIV = "CIV",
    HRV = "HRV",
    CUB = "CUB",
    CYP = "CYP",
    COD = "COD",
    DJI = "DJI",
    ECU = "ECU",
    SLV = "SLV",
    GNQ = "GNQ",
    ERI = "ERI",
    EST = "EST",
    ETH = "ETH",
    FLK = "FLK",
    FRO = "FRO",
    FJI = "FJI",
    FIN = "FIN",
    GUF = "GUF",
    PYF = "PYF",
    ATF = "ATF",
    GAB = "GAB",
    GEO = "GEO",
    GHA = "GHA",
    GIB = "GIB",
    GRC = "GRC",
    GRL = "GRL",
    GLP = "GLP",
    GUM = "GUM",
    GTM = "GTM",
    GGY = "GGY",
    GIN = "GIN",
    GNB = "GNB",
    GCC = "GCC",
    GUY = "GUY",
    HTI = "HTI",
    HMD = "HMD",
    HND = "HND",
    ISL = "ISL",
    IRN = "IRN",
    IRQ = "IRQ",
    IMN = "IMN",
    ISR = "ISR",
    JEY = "JEY",
    KAZ = "KAZ",
    KIR = "KIR",
    KGZ = "KGZ",
    LVA = "LVA",
    LSO = "LSO",
    LBR = "LBR",
    LBY = "LBY",
    LIE = "LIE",
    LTU = "LTU",
    MAC = "MAC",
    MDG = "MDG",
    MWI = "MWI",
    MDV = "MDV",
    MLI = "MLI",
    MLT = "MLT",
    MHL = "MHL",
    MTQ = "MTQ",
    MRT = "MRT",
    MUS = "MUS",
    MYT = "MYT",
    FSM = "FSM",
    MDA = "MDA",
    MCO = "MCO",
    MNG = "MNG",
    MNE = "MNE",
    MSR = "MSR",
    MOZ = "MOZ",
    NAM = "NAM",
    NRU = "NRU",
    NPL = "NPL",
    NCL = "NCL",
    NIC = "NIC",
    NER = "NER",
    NGA = "NGA",
    NIU = "NIU",
    NFK = "NFK",
    PRK = "PRK",
    MNP = "MNP",
    PAK = "PAK",
    PLW = "PLW",
    PAN = "PAN",
    PNG = "PNG",
    PRY = "PRY",
    PER = "PER",
    PCN = "PCN",
    PRT = "PRT",
    QAT = "QAT",
    MKD = "MKD",
    COG = "COG",
    REU = "REU",
    ROU = "ROU",
    RUS = "RUS",
    RWA = "RWA",
    BLM = "BLM",
    SHN = "SHN",
    KNA = "KNA",
    LCA = "LCA",
    MAF = "MAF",
    SPM = "SPM",
    VCT = "VCT",
    WSM = "WSM",
    SMR = "SMR",
    STP = "STP",
    SEN = "SEN",
    SRB = "SRB",
    SYC = "SYC",
    SLE = "SLE",
    SXM = "SXM",
    SVK = "SVK",
    SVN = "SVN",
    SLB = "SLB",
    SOM = "SOM",
    ZAF = "ZAF",
    SGS = "SGS",
    SSD = "SSD",
    LKA = "LKA",
    PSE = "PSE",
    SDN = "SDN",
    SUR = "SUR",
    SJM = "SJM",
    SWZ = "SWZ",
    SYR = "SYR",
    TJK = "TJK",
    TZA = "TZA",
    GMB = "GMB",
    TLS = "TLS",
    TGO = "TGO",
    TKL = "TKL",
    TON = "TON",
    TUN = "TUN",
    TKM = "TKM",
    TUV = "TUV",
    UGA = "UGA",
    UMI = "UMI",
    URY = "URY",
    UZB = "UZB",
    VUT = "VUT",
    VAT = "VAT",
    VEN = "VEN",
    VGB = "VGB",
    WLF = "WLF",
    ESH = "ESH",
    YEM = "YEM",
    ZMB = "ZMB",
    ZWE = "ZWE"
}

export enum Alpha2 {
    US = "US",
    PR = "PR",
    CA = "CA",
    GB = "GB",
    AE = "AE",
    AI = "AI",
    AT = "AT",
    AU = "AU",
    BB = "BB",
    BE = "BE",
    BH = "BH",
    BQ = "BQ",
    BS = "BS",
    CH = "CH",
    CO = "CO",
    CW = "CW",
    CZ = "CZ",
    DE = "DE",
    DK = "DK",
    DM = "DM",
    DO = "DO",
    DZ = "DZ",
    EG = "EG",
    ES = "ES",
    FR = "FR",
    GD = "GD",
    HU = "HU",
    IE = "IE",
    IN = "IN",
    IT = "IT",
    JO = "JO",
    JP = "JP",
    KE = "KE",
    KR = "KR",
    KW = "KW",
    LB = "LB",
    LU = "LU",
    MX = "MX",
    NL = "NL",
    NO = "NO",
    NZ = "NZ",
    OM = "OM",
    PH = "PH",
    PL = "PL",
    SA = "SA",
    SE = "SE",
    SG = "SG",
    TC = "TC",
    TH = "TH",
    TR = "TR",
    TT = "TT",
    UA = "UA",
    VI = "VI",
    MA = "MA",
    LA = "LA",
    MY = "MY",
    MM = "MM",
    TW = "TW",
    VN = "VN",
    HK = "HK",
    KH = "KH",
    ID = "ID",
    JM = "JM",
    AF = "AF",
    AX = "AX",
    AL = "AL",
    AS = "AS",
    AD = "AD",
    AO = "AO",
    AQ = "AQ",
    AG = "AG",
    AR = "AR",
    AM = "AM",
    AW = "AW",
    AZ = "AZ",
    BD = "BD",
    BY = "BY",
    BZ = "BZ",
    BJ = "BJ",
    BM = "BM",
    BT = "BT",
    BO = "BO",
    BA = "BA",
    BW = "BW",
    BV = "BV",
    BR = "BR",
    IO = "IO",
    BN = "BN",
    BG = "BG",
    BF = "BF",
    BI = "BI",
    CM = "CM",
    CV = "CV",
    KY = "KY",
    CF = "CF",
    TD = "TD",
    CL = "CL",
    CN = "CN",
    CX = "CX",
    CC = "CC",
    KM = "KM",
    CK = "CK",
    CR = "CR",
    CI = "CI",
    HR = "HR",
    CU = "CU",
    CY = "CY",
    CD = "CD",
    DJ = "DJ",
    EC = "EC",
    SV = "SV",
    GQ = "GQ",
    ER = "ER",
    EE = "EE",
    ET = "ET",
    FK = "FK",
    FO = "FO",
    FJ = "FJ",
    FI = "FI",
    GF = "GF",
    PF = "PF",
    TF = "TF",
    GA = "GA",
    GE = "GE",
    GH = "GH",
    GI = "GI",
    GR = "GR",
    GL = "GL",
    GP = "GP",
    GU = "GU",
    GT = "GT",
    GG = "GG",
    GN = "GN",
    GW = "GW",
    GC = "GC",
    GY = "GY",
    HT = "HT",
    HM = "HM",
    HN = "HN",
    IS = "IS",
    IR = "IR",
    IQ = "IQ",
    IM = "IM",
    IL = "IL",
    JE = "JE",
    KZ = "KZ",
    KI = "KI",
    KG = "KG",
    LV = "LV",
    LS = "LS",
    LR = "LR",
    LY = "LY",
    LI = "LI",
    LT = "LT",
    MO = "MO",
    MG = "MG",
    MW = "MW",
    MV = "MV",
    ML = "ML",
    MT = "MT",
    MH = "MH",
    MQ = "MQ",
    MR = "MR",
    MU = "MU",
    YT = "YT",
    FM = "FM",
    MD = "MD",
    MC = "MC",
    MN = "MN",
    ME = "ME",
    MS = "MS",
    MZ = "MZ",
    NA = "NA",
    NR = "NR",
    NP = "NP",
    NC = "NC",
    NI = "NI",
    NE = "NE",
    NG = "NG",
    NU = "NU",
    NF = "NF",
    KP = "KP",
    MP = "MP",
    PK = "PK",
    PW = "PW",
    PA = "PA",
    PG = "PG",
    PY = "PY",
    PE = "PE",
    PN = "PN",
    PT = "PT",
    QA = "QA",
    MK = "MK",
    CG = "CG",
    RE = "RE",
    RO = "RO",
    RU = "RU",
    RW = "RW",
    BL = "BL",
    SH = "SH",
    KN = "KN",
    LC = "LC",
    MF = "MF",
    PM = "PM",
    VC = "VC",
    WS = "WS",
    SM = "SM",
    ST = "ST",
    SN = "SN",
    RS = "RS",
    SC = "SC",
    SL = "SL",
    SX = "SX",
    SK = "SK",
    SI = "SI",
    SB = "SB",
    SO = "SO",
    ZA = "ZA",
    GS = "GS",
    SS = "SS",
    LK = "LK",
    PS = "PS",
    SD = "SD",
    SR = "SR",
    SJ = "SJ",
    SZ = "SZ",
    SY = "SY",
    TJ = "TJ",
    TZ = "TZ",
    GM = "GM",
    TL = "TL",
    TG = "TG",
    TK = "TK",
    TO = "TO",
    TN = "TN",
    TM = "TM",
    TV = "TV",
    UG = "UG",
    UM = "UM",
    UY = "UY",
    UZ = "UZ",
    VU = "VU",
    VA = "VA",
    VE = "VE",
    VG = "VG",
    WF = "WF",
    EH = "EH",
    YE = "YE",
    ZM = "ZM",
    ZW = "ZW"
}

export enum Currency {
    USD = "USD",
    CAD = "CAD",
    MXN = "MXN",
    GBP = "GBP",
    EUR = "EUR",
    CHF = "CHF",
    COP = "COP",
    JPY = "JPY",
    PHP = "PHP"
}

export enum optInNotificationTypes {
    EMAIL = "EMAIL",
    SMS = "SMS"
}
