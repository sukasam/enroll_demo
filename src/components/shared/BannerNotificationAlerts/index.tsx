/** @jsxImportSource @emotion/react */
import { useTranslate } from "Components/shared/Translate";
import { Alpha2 } from "Constants/countryConfig/enums";
import { BasicBanner, EmptyBasicBanner } from "Services/launchDarkly/types";
import isActiveBanner from "Services/launchDarkly/utiles";
import styles from "./styles";

interface BannerProps {
    currentMarket: Alpha2;
    rosettaAlterations: (message: string, activeFlag: BasicBanner) => string;
    bannerFlagLD: EmptyBasicBanner;
    children?: JSX.Element | never;
}

export default function BannerNotificationAlerts(
    props: BannerProps
): JSX.Element | null {
    const { currentMarket, rosettaAlterations, bannerFlagLD, children } = props;
    const translate = useTranslate();

    if (!isActiveBanner(bannerFlagLD)) {
        return null;
    }
    const activeBannerFlagLD = bannerFlagLD as BasicBanner;

    if (!activeBannerFlagLD.markets.includes(currentMarket)) {
        return null;
    }

    let displayMessage = translate(activeBannerFlagLD.rosettaTag);
    displayMessage = rosettaAlterations(displayMessage, activeBannerFlagLD);

    return (
        <div css={styles}>
            <div className="container">
                <div className="banner">
                    <div
                        className="message"
                        dangerouslySetInnerHTML={{
                            __html: displayMessage
                        }}
                    />
                    <div className="children">{children}</div>
                </div>
            </div>
        </div>
    );
}
