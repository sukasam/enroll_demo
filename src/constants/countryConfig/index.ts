import { useTranslations } from "Contexts/translation";
import { useMemo } from "react";
import countries from "./countries";
import { Alpha2 } from "./enums";
import {
    CountryConfig,
    Locale,
    SupportedCountryConfig,
    UnsupportedCountryConfig
} from "./types";

// This makes importing less confusing
export { countries as countryConfigs };

export function getSuppotedCountries(): SupportedCountryConfig[] {
    return countries.filter<SupportedCountryConfig>(
        (config): config is SupportedCountryConfig => config.isSupported
    );
}

export function getUnsupportedCountries(): UnsupportedCountryConfig[] {
    return countries.filter(
        (config): config is UnsupportedCountryConfig => !config.isSupported
    );
}

export function getCountries(supported: boolean): CountryConfig[] {
    if (supported) {
        return getSuppotedCountries();
    }

    return countries;
}

export function getCountryConfig(
    countryCode: Alpha2,
    supportedOnly: true
): SupportedCountryConfig | undefined;

export function getCountryConfig(
    countryCode: Alpha2,
    supportedOnly: false
): CountryConfig | undefined;

export function getCountryConfig(
    countryCode: Alpha2,
    supportedOnly: boolean
): CountryConfig | undefined {
    if (supportedOnly) {
        return getSuppotedCountries().find(
            config => config.alpha2 === countryCode
        );
    }

    return countries.find(config => config.alpha2 === countryCode);
}

export function isSupported(
    config: CountryConfig
): config is SupportedCountryConfig {
    return config.isSupported;
}

export function useCountryConfig(): SupportedCountryConfig | undefined {
    const { country } = useTranslations();
    const alpha2 = country as Alpha2;
    const config = useMemo(() => getCountryConfig(alpha2, true), [alpha2]);
    return config;
}

export function getDefaultLanguage(alpha2: Alpha2): string {
    const config = getCountryConfig(alpha2, true);

    return config?.languages.find(lang => lang.default)?.code ?? "en";
}

export function getValidatedLocale(
    countryCode: string | null | Alpha2,
    language: string | null
): Locale {
    // Default to US if no country or invalid country
    const countryConfig =
        getCountryConfig(countryCode as Alpha2, true) ||
        getCountryConfig(Alpha2.US, true);

    // Get default language for the country
    const defaultLanguage = getDefaultLanguage(countryCode as Alpha2);

    // Validate the provided language against country's available languages
    const isValidLanguage =
        language &&
        countryConfig?.languages?.some(lang => lang.code === language);

    return {
        countryCode: countryConfig?.alpha2 || Alpha2.US,
        language: isValidLanguage ? language! : defaultLanguage
    };
}
