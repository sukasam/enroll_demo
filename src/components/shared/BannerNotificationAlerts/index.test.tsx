import { Alpha2 } from "Constants/countryConfig/enums";
import { BasicBanner } from "Services/launchDarkly/types";
import { render } from "Shared/testUtils";
import BannerNotificationAlerts from "./index";

// Mock the useTranslate hook
jest.mock("Components/shared/Translate", () => ({
    useTranslate: (): ((key: string) => string) => (key: string) => key
}));

describe("BannerNotificationAlerts", () => {
    it("renders correctly", () => {
        const props = {
            currentMarket: Alpha2.US,
            rosettaAlterations: (message: string): string => message,
            bannerFlagLD: {
                markets: [Alpha2.US],
                rosettaTag: "test.message"
            } as BasicBanner
        };

        const { container } = render(<BannerNotificationAlerts {...props} />);
        expect(container).toMatchSnapshot();
    });
});
