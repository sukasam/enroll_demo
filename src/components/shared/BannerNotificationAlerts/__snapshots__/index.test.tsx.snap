// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerNotificationAlerts renders correctly 1`] = `
<div>
  <div
    class="css-nmzf5j"
  >
    <div
      class="container"
    >
      <div
        class="banner"
      >
        <div
          class="message"
        >
          test.message
        </div>
        <div
          class="children"
        />
      </div>
    </div>
  </div>
</div>
`;
